src/util/tile/fd_tile_threads.cxx

```diff
@@ -526,177 +567,250 @@ fd_tile_private_cpus_parse( char const * cstr,
-    FD_VOLATILE( args->idx      ) = tile_idx;
-    FD_VOLATILE( args->cpu_idx  ) = cpu_idx;
-    FD_VOLATILE( args->stack    ) = stack;
-    FD_VOLATILE( args->stack_sz ) = stack_sz;
-    FD_VOLATILE( args->tile     ) = NULL;
+static int
+fd_tile_private_group_manager( void * _args ) {
+  fd_tile_private_group_manager_args_t * args = (fd_tile_private_group_manager_args_t *)_args;
+
+  for( ulong i = 0; i < args->group_id; i++ ) {
+    /* The parent group launcher is mapping task dispatch areas one by one before
+       spawning tiles, but we don't want tiles to be able to dispatch to tiles outside
+       their group, so unmap these inside the new process. Seccomp will prevent it
+       from being remapped again.  */
+    if( FD_LIKELY( args->mmap[i] ) ) {
+      if( FD_UNLIKELY( munmap( args->mmap[ i ], args->mmap_sz[ i ] ) ) )
+        FD_LOG_ERR(( "fd_tile: munmap failed (%i-%s)", errno, strerror( errno ) ));
+    }
+  }
 
-    FD_COMPILER_MFENCE();
+  fd_memcpy( fd_tile_private_cpu_id, args->tile_to_cpu+fd_tile_private_id0, fd_tile_private_cnt*sizeof(ushort) );
 
-    err = pthread_create( &fd_tile_private[tile_idx].pthread, attr, fd_tile_private_manager, args );
-    if( FD_UNLIKELY( err ) ) {
-      if( fixed ) FD_LOG_ERR(( "fd_tile: pthread_create failed (%i-%s)\n\t"
-                               "Unable to start up the tile %lu on cpu %lu.  Likely causes for this include\n\t"
-                               "this cpu is restricted from the user or does not exist on this host.\n\t"
-                               "Update --tile-cpus to specify a set of allowed cpus that have been reserved\n\t"
-                               "for this thread group on this host.",
-                               err, strerror( err ), tile_idx, cpu_idx ));
-      FD_LOG_ERR(( "fd_tile: pthread_create failed (%i-%s)\n\tUnable to start up the tile %lu (floating).",
-                   err, strerror( err ), tile_idx ));
-    }
+  for( ulong i = 1; i < fd_tile_private_cnt; i++ ) {
+      fd_tile_private_boot_single( i, args->host_id, args->tile_to_cpu, args->floating_cpu_set, 0 );
+  }
 
-    /* Wait for the tile to be ready to exec */
+  fd_tile_private_manager_args_t new_args[1];
+  FD_VOLATILE( new_args->id       ) = fd_tile_private_id0;
+  FD_VOLATILE( new_args->idx      ) = 0;
+  FD_VOLATILE( new_args->cpu_idx  ) = args->tile_to_cpu[ 0 ];
+  FD_VOLATILE( new_args->stack    ) = args->stack;
+  FD_VOLATILE( new_args->stack_sz ) = FD_TILE_PRIVATE_STACK_SZ;
+  FD_VOLATILE( new_args->tile     ) = NULL;
+  FD_COMPILER_MFENCE();
 
-    fd_tile_private_t * tile;
-    for(;;) {
-      tile = FD_VOLATILE_CONST( args->tile );
-      if( FD_LIKELY( tile ) ) break;
-      FD_YIELD();
-    }
-    FD_VOLATILE( fd_tile_private[ tile_idx ].tile ) = tile;
-    FD_VOLATILE( fd_tile_private[ tile_idx ].lock ) = tile;
+  fd_tile_private_manager( new_args );
+
+  // Finish sandboxing in any cloned processes
+  fd_sandbox_private();
```

Comment:
This should happen before we mark all the tiles in this process as ready, else it's possible in a weird race / hang situation a tile could get dispatched to before it's fully sandboxed.